dev/Kozani/KozaniManager/main.cpp

```diff
@@ -2,28 +2,82 @@
 // Licensed under the MIT License.
 
 #include "pch.h"
+#include <fstream>
+#include <filesystem>
 
 #include "..\KozaniManager\KozaniManager-Constants.h"
 
 #include <KozaniManager_h.h>
 
-#include "Microsoft.Kozani.Activation.h"
-
 // Including this file once per binary will automatically opt WIL error handling macros into calling RoOriginateError when they
 // begin logging a new error.  This greatly improves the debuggability of errors that propagate before a failfast.
 #include <wil/result_originate.h>
 
 #include <wrl\module.h>
 
 #include <windows.applicationmodel.activation.h>
+#include "KozaniDvc-Constants.h"
+#include "ConnectionManager.h"
 
 using namespace Microsoft::WRL;
+using namespace Microsoft::Kozani::Manager;
+
+namespace Dvc = Microsoft::Kozani::DVC;
 
 // Implement the LifetimeManager as a classic COM Out-of-Proc server, via WRL
 // See https://docs.microsoft.com/cpp/cppcx/wrl/how-to-create-a-classic-com-component-using-wrl?redirectedfrom=MSDN&view=vs-2019 for more details
 
 static constexpr GUID KozaniManager_guid { PR_KOZANIMANAGER_CLSID_GUID };
 
+Microsoft::Kozani::Manager::ConnectionManager g_connectionManager;
+
+// Accumulated number of connections to remote server. Used to keep track of whether a new connection is made between remote desktop client is
+// launched and connection wait time out.
+volatile LONG g_newConnectionCount{};
+
+static std::string GetConnectionId(PCWSTR connectionRdpFilePath)
+{
+    auto fileStatus{ std::filesystem::status(connectionRdpFilePath) };
+
+    // Specifically checks file existence and file type so we can throw specific error code instead of letting the std::fstream fail later and throw 
+    // generic error code ERROR_UNHANDLED_EXCEPTION for better error messaging.
```

Comment:
Changed to try/catch. Error message makes it easier to read in the logs. Otherwise, we need to check back in the code to know what the error indicates.